src/BikeTag/AppDelegate.swift

```diff
@@ -36,6 +37,35 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     return true
   }
 
+  func application(application: UIApplication, didRegisterUserNotificationSettings notificationSettings: UIUserNotificationSettings) {
+    if notificationSettings.types != .None {
+      application.registerForRemoteNotifications()
+    }
+  }
+
+  func application(application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: NSData) {
+    let tokenChars = UnsafePointer<CChar>(deviceToken.bytes)
+    var tokenString = ""
+
+    for i in 0..<deviceToken.length {
+      tokenString += String(format: "%02.2hhx", arguments: [tokenChars[i]])
+    }
+
+    let logSuccess = {
+      Logger.info("Successfully registered for remote notifications with device token: \(tokenString)")
+    }
+
+    let logError = { (error: NSError) -> () in
+      Logger.error("User registered for notifications with device token \(tokenString), but failed to submit to API: \(error)")
+    }
+
+    DevicesService().postNewDevice(tokenString, successCallback: logSuccess, errorCallback: logError)
```

Comment:
Yeah... "it depends" on the call.

"Connectivity" as a status is really only part of the picture. I think what we really care about is if the request succeeds or not. Connectivity is only one of the reasons it could fail.

So, if it's an essential request (like getting your API credentials) I have a ["repeat until it succeeds" dialog](https://github.com/jackpine/biketag-ios/blob/master/src/BikeTag/Controllers/HomeViewController.swift#L126).

I decided this particular request (Device registration) was non-essential, and displaying an alert was more damage than it was worth. Rather, it just means the user won't get push notifications if someone captures their spot. Furthermore, once the user has told us it's OK to receive notifications, we re-submit their device token every time they launch the app to keep it in sync, this way we'll eventually recover from any intermittent failure.